### PR TITLE
Fixed compiler error in dash::mismatch

### PR DESCRIPTION
**Pull request by fuchsto**

### CODE REVIEW

File: dash/include/dash/algorithm/Accumulate.h [hunk 1]
```diff
@@ -82,21 +83,22 @@ ValueType accumulate(
...
   auto l_first        = index_range.begin;
   auto l_last         = index_range.end;
   auto l_result = std::accumulate(l_first, l_last, init, binary_op);
 
-  dash::Array<index_t> l_results(dash::size());
+  dash::Array<index_t> l_results(team.size());
 
   l_results.local[0] = l_result;
 
   in_first.pattern().team().barrier();
```
Comment (devreal) [inline dash/include/dash/algorithm/Accumulate.h]: I assume `in_first.pattern().team().barrier()` is the same as `team.barrier()` here? A unified use would improve readability.

**Review by devreal — CHANGES_REQUESTED** [review]
Usage of `team.barrier()` vs `in_first.pattern().team().barrier()` should be unified

**Comment by fuchsto** [comment]
@devreal Done, also fixed missing `#endif` in `algorithm/Mismatch.h` 

@JosefSchaeffer There is no unit test suite for `dash::mismatch`, please add it!

**Review by devreal — APPROVED** [review]

**Comment by fuchsto** [comment]
Merging after feedback from @devreal

**Pull request merged**